src/Storage/JsonFileFeed.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * JSON File Feed class.
+ *
+ * @package Automattic\WooCommerce\ProductFeedForOpenAI
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\ProductFeedForOpenAI\Storage;
+
+use Automattic\WooCommerce\ProductFeedForOpenAI\Feed\FeedInterface;
+
+// This file works directly with local files. That's fine.
+// phpcs:disable WordPress.WP.AlternativeFunctions
+
+/**
+ * In-memory feed storage.
+ *
+ * This class simply stores all data in memory and generates JSON at once.
+ */
+class JsonFileFeed implements FeedInterface {
+	/**
+	 * Indicates if there are previous entries in the feed.
+	 *
+	 * @var bool
+	 */
+	private $has_entries = false;
+
+	/**
+	 * The path to the feed file.
+	 *
+	 * @var string
+	 */
+	private $file_path;
+
+	/**
+	 * The file handle.
+	 *
+	 * @var resource|null
+	 */
+	private $file_handle = null;
+
+	/**
+	 * Start the feed.
+	 *
+	 * @return void
+	 */
+	public function start(): void {
+		$upload_dir = wp_upload_dir( null, true );
+		$directory  = $upload_dir['basedir'] . '/product-feeds/';
+
+		if ( ! is_dir( $directory ) ) {
+			mkdir( $directory, 0755, true ); // We need to reconsider this access.
```

Comment:
Should probably use https://developer.wordpress.org/reference/functions/wp_mkdir_p/